src/ui/win32/DialogBase.cpp

```diff
@@ -99,13 +139,23 @@ BOOL DialogBase::OnCommand(WORD nCommand)
     {
         case IDOK:
             m_vmWindow.SetDialogResult(ra::ui::DialogResult::OK);
-            DestroyWindow(m_hWnd);
+
+            if (m_bModal)
+                EndDialog(m_hWnd, IDOK);
+            else
+                DestroyWindow(m_hWnd);
+
             return TRUE;
 
         case IDCLOSE:
         case IDCANCEL:
             m_vmWindow.SetDialogResult(ra::ui::DialogResult::Cancel);
-            DestroyWindow(m_hWnd);
+
+            if (m_bModal)
+                EndDialog(m_hWnd, nCommand);
+            else
+                DestroyWindow(m_hWnd);
```

Comment:
Not sure if it should be done here but the handle should be set to nullptr after being destroyed since the handle us invalid now.